src/app/icd/BUILD.gn

```diff
@@ -13,9 +13,30 @@
 # limitations under the License.
 
 import("//build_overrides/chip.gni")
+import("${chip_root}/build/chip/buildconfig_header.gni")
 import("icd.gni")
 
 # ICD Server sources and configurations
+
+buildconfig_header("icd_buildconfig") {
+  header = "ICDBuildConfig.h"
+  header_dir = "app/icd"
+
+  defines = [
+    "CHIP_CONFIG_ENABLE_ICD_SERVER=${chip_enable_icd_server}",
+    "ICD_REPORT_ON_ENTER_ACTIVE_MODE=${chip_report_on_active_mode}",
+    "ICD_MAX_NOTIFICATION_SUBSCRIBERS=${icd_max_notification_subscribers}",
+  ]
+
+  visibility = [ ":icd_config" ]
+}
+
+source_set("icd_config") {
+  sources = [ "ICDConfig.h" ]
```

Comment:
As I understand it this creates 2 headers` ICDBuildConfig.h` and `ICDConfig.h` where ICDConfig includes the latter.

Does it need to be like this? Can't it be only `ICDBuildConfig` and be renamed `ICDConfig`

Reply:
Didn't find the explanation but all the *BuildConfig have the same structure so kept it inline.

Reply:
I don't think you need this ... just keep BuildConfig for now as there are no other things. Intermediate seems only needed if you have extra configurations.

Reply:
Pinged @mspang about this - it does seem intentional after all.

Reply:
Basically this may be a remnant of automake days ... however because we already use the pattern in many places, lets only change it as a whole to not be inconsistent.

Could you add an issue for changing this pattern globally? (but keep it in this PR for consistency)

Reply:
Follow up issue to clean this up at some point
https://github.com/project-chip/connectedhomeip/issues/30687